module-system/sov-modules-api/src/lib.rs

```diff
@@ -290,3 +300,45 @@ pub trait RpcRunner {
     type Context: Context;
     fn get_storage(&self) -> <Self::Context as Spec>::Storage;
 }
+
+/// Sorts ModuleInfo objects by their dependencies
+pub fn sort_modules_by_dependencies<'a, C: Context, T>(
```

Comment:
Question: Why do we use a generic `T` here? Can we pick a suitable integer type (maybe usize) instead, or is there some blocker?

Reply:
So I simplified the function to:
```rust
pub fn sort_modules_by_dependencies<C: Context>(
    modules: Vec<&dyn ModuleInfo<Context = C>>,
) -> Result<Vec<&dyn ModuleInfo<Context = C>>, anyhow::Error> 
```
But I also kept the mapping one and it calls the sort_modules_by_dependencies
```rust
pub fn sort_values_by_modules_dependencies<'a, C: Context, TValue>(
    module_value_tuples: Vec<(&dyn ModuleInfo<Context = C>, &'a TValue)>,
) -> Result<Vec<&'a TValue>, anyhow::Error> where TValue: 'a 
```
I want to have the second one because it is easier to call it from the macro generated code rather than generate that code inside the macro.